### PR TITLE
Remove CommandLineParser dependency from the profiler libraries (#18)

### LINKED ISSUE #18: Is it possible to not dependent on CommandLineParser for the core library?
Is it possible to not dependent on `command line parser`?

https://github.com/Azure/azuremonitor-opentelemetry-profiler-net/blob/06809fbbf8bc7f683b37c862b80fb3f583645718/src/ServiceProfiler.EventPipe.Otel/Azure.Monitor.OpenTelemetry.Profiler.Core/Azure.Monitor.OpenTelemetry.Profiler.Core.nuspec#L21C1-L22C1

Uploader is using that to parsing the command line but we shall not need it for the Core.

### PR DESCRIPTION
**Pull request by xiaomi7732**
Fixes #18.

## Summary

The third-party [`CommandLineParser`](https://www.nuget.org/packages/CommandLineParser) package was only ever needed by the standalone uploader to **parse** its command line — yet it was referenced by the profiler libraries too. This PR removes the dependency entirely from the main repo and replaces the uploader's argument parsing with the already-referenced `Microsoft.Extensions.Configuration` command-line binding.

## What changed

- **Producer** (`UploadContextModel.ToString()`, in the Shared library): now emits `--<PropertyName> "value"` keys that match the uploader's `UploadContext` property names so the configuration binder maps them directly. Boolean flags are emitted as `--Flag true` only when set.
- **Consumer** (`ServiceProfiler.EventPipe.Upload`):
  - `UploadContext` drops the `[Option]` attributes and `using CommandLine`; it's now a plain bindable options class.
  - `Program.Main` binds via `ConfigurationBuilder().AddCommandLine(args).Get<UploadContext>()`. It **fails fast** with exit code `1` (and a clear `fail:` message) on malformed arguments, and returns `Environment.ExitCode` after the host completes so the uploader's success (`0`) / failure (`-1`) exit code is preserved for the parent process.
  - Added `Microsoft.Extensions.Configuration.Binder` (drives `Get<T>()`).
- **Removed** the dead `CommandLineParser` `<PackageReference>` from the Core library, the public `Azure.Monitor.OpenTelemetry.Profiler` package, and the classic `ServiceProfiler.EventPipe.Client` project (none of them used it).

## Tests

- Updated the producer-format assertions in `UploadContextModelTests`.
- Added `UploadContextRoundTripTests`, which serializes via `UploadContextModel.ToString()`, tokenizes using the **real Windows `CommandLineToArgvW`** parser (matching how `ProcessStartInfo.Arguments` is parsed by the launched uploader), then binds back into `UploadContext` — verifying every field (Guid, Uri, DateTimeOffset, enum, bool flags, optional/nullable strings).

## Verification

- Combined solution builds with 0 errors.
- Affected test suites pass: Uploader (24), classic Client (58), OTel (70).
- Clean build + pack of the OTel NuGet package confirms `CommandLineParser` is **no longer a dependency** in the generated `.nuspec`, with `Microsoft.Extensions.Configuration.Binder` correctly present.

## Review

Iteratively reviewed with GPT-5.5 and Opus 4.7 to **two consecutive clean rounds**. Findings that led to real improvements: the round-trip test now tokenizes via the real Windows `CommandLineToArgvW` (instead of an approximation), `Program.Main` fails fast on malformed args, and `Main` returns `Environment.ExitCode` so uploader failures are no longer masked by a literal `0`.

## Notes

- Producer and consumer ship together in the same package, so the internal command-line wire format can change freely; no cross-version compatibility contract exists for these args.
- The root `Directory.Packages.props` is a symlink into the `ServiceProfiler` submodule whose `CommandLineParser` version is still used by submodule projects, so the central `PackageVersion` entry is intentionally left in place (an unreferenced version never becomes a package dependency).

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                $@" --{nameof(UploadContext.HostUrl)} ""https://example.endpoint.com/""" +
+                $@" --{nameof(UploadContext.MetadataFilePath)} ""C:\traces\meta data.json""" +
+                $@" --{nameof(UploadContext.UploadMode)} ""{UploadMode.Always}""" +
+                $@" --{nameof(UploadContext.SerializedSampleFilePath)} ""C:\traces\samples file.json""" +
+                $@" --{nameof(UploadContext.PipeName)} ""pipe-name-123""" +
+                $@" --{nameof(UploadContext.PreserveTraceFile)} true" +
+                $@" --{nameof(UploadContext.SkipEndpointCertificateValidation)} true" +
+                $@" --{nameof(UploadContext.RoleName)} ""my role name""" +
+                $@" --{nameof(UploadContext.TriggerType)} ""my trigger""" +
+                $@" --{nameof(UploadContext.TraceFileFormat)} ""Nettrace""";
```
Comment (CarlosCampos06) [inline tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs]: Before this merges, I'd like the new `UploadContextRoundTripTests` to actually drive the real producer. Right now it hand-builds the command line in the test instead of calling `UploadContextModel.ToString()`, so what we're really verifying is that the binder can parse a string we wrote in the test — not that the producer's output binds to the consumer, which is the whole contract this PR is changing. If the producer and consumer ever drift (key names, quoting, bool emission, defaults, timestamp format), the test would still pass.

Can we construct a real `UploadContextModel`, call `.ToString()`, tokenize it through the same `CommandLineToArgvW` path, bind it back into `UploadContext`, and assert every field round-trips (including the optional-fields-omitted/defaults case)? That makes it a genuine round-trip and gets the path-quoting edge cases covered for free.

File: tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        [Fact]
+        public void CommandLine_BindsToUploadContext_AllFields()
+        {
+            Guid iKey = Guid.NewGuid();
+            DateTimeOffset sessionId = new(2024, 6, 29, 13, 45, 30, TimeSpan.Zero);
+
+            string commandLine =
+                $@"--{nameof(UploadContext.TraceFilePath)} ""C:\traces\my trace.nettrace""" +
+                $@" --{nameof(UploadContext.AIInstrumentationKey)} ""{iKey}""" +
+                $@" --{nameof(UploadContext.SessionId)} ""{sessionId.UtcDateTime:o}""" +
```
Comment (CarlosCampos06) [inline tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs]: Following on from the round-trip test: the producer serializes `SessionId` with `TimestampContract.TimestampToString(...)`, but the test uses `:o`, so we're not actually proving the real timestamp format survives the trip. What I'm worried about is the invariant `DateTimeOffset` binder parsing a *different* instant (e.g. if the serialized form has no offset) — and because the runtime validator only checks `SessionId != default`, a shifted-but-non-default value would pass validation and then corrupt the blob `StartTime`/`TriggerTime`. If we drive the real `UploadContextModel.ToString()` in the test and assert `model.SessionId == bound.SessionId`, that pins it down.

File: src/ServiceProfiler.EventPipe.Upload/UploadContext.cs [hunk 3]
```diff
@@ -2,117 +2,41 @@
...
         public Uri HostUrl { get; set; } = null!;
 
-        [Option(SessionIdKeyName, Required = true, HelpText = "Trace session id.")]
         public DateTimeOffset SessionId { get; set; }
 
-        [Option(StampIdShortKeyName, StampIdKeyName, Required = true, HelpText = "StampId used to upload the trace file.")]
         public string StampId { get; set; } = null!;
 
-        [Option(TraceFilePathShortKeyName, TraceFilePathKeyName, Required = true, HelpText = "File path to trace file.")]
         public string TraceFilePath { get; set; } = null!;
```
Comment (CarlosCampos06) [inline src/ServiceProfiler.EventPipe.Upload/UploadContext.cs:20]: Minor one: now that the `[Option(Required = true)]` attributes are gone, `TraceFilePath`, `StampId`, and `TraceFileFormat` aren't enforced anywhere — the runtime `UploadContextValidator` doesn't check them. In practice the producer always sets them so it's not urgent, but I'd feel better if we at least guarded `TraceFilePath` in `UploadContextValidator`, or dropped a comment noting the missing required-validation is intentional.

File: tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs (hunk 1 above)
Comment (xiaomi7732) [reply tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs]: Done in 43ca505. The test now constructs a real `UploadContextModel`, calls the actual `UploadContextModel.ToString()`, tokenizes it through the same `CommandLineToArgvW` path, and binds back into `UploadContext`, asserting every field round-trips (plus an optional-fields-omitted/defaults case). The producer source is linked into the test assembly since the uploader assembly deliberately doesn't reference this producer-only type. Now real producer/consumer drift (keys, quoting, bool emission, defaults, timestamp format) would fail the test.

File: tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs (hunk 2 above)
Comment (xiaomi7732) [reply tests/ServiceProfiler.EventPipe.Upload.Tests/UploadContextRoundTripTests.cs]: Fixed in 43ca505. The round-trip now drives the real producer (which serializes via `TimestampContract.TimestampToString`) and asserts `model.SessionId == bound.SessionId` using a session id with sub-second precision, so an offset-less/lossy serialization that shifted the instant would fail. (For the record, `TimestampToString` formats `UtcDateTime` with `"o"`, i.e. it includes the `Z` offset and full ticks, so the instant round-trips exactly.)

File: src/ServiceProfiler.EventPipe.Upload/UploadContext.cs (hunk 3 above)
Comment (xiaomi7732) [reply src/ServiceProfiler.EventPipe.Upload/UploadContext.cs:20]: Addressed in 43ca505: restored required validation for `TraceFilePath` in the uploader-side `UploadContextValidator` so a missing trace path fails fast with a clear message instead of an obscure downstream file error, with tests covering null/empty. I left `StampId`/`TraceFileFormat` unguarded (`TraceFileFormat` has a non-null default on both ends; `StampId` isn't consumed as a path) - happy to add those too if you'd prefer belt-and-suspenders.

**Review by CarlosCampos06 — APPROVED** [review]
All three points addressed in 43ca505:

1. `UploadContextRoundTripTests` now constructs a real `UploadContextModel`, calls the actual `.ToString()`, tokenizes through `CommandLineToArgvW` and binds back into `UploadContext` - a genuine producer->consumer round trip that would catch key/quoting/bool/default/timestamp drift.
2. Timestamp pinned via `Assert.Equal(model.SessionId, bound.SessionId)` with sub-second precision, so a lossy/offset-less form that shifts the instant fails.
3. `TraceFilePath` required-validation restored in `UploadContextValidator` with null/empty test coverage.

Leaving `StampId`/`TraceFileFormat` unguarded is reasonable given the defaults/usage. LGTM.
